conf/upsd.conf.sample

```diff
@@ -64,6 +64,12 @@
 # Note that it is not true for Windows platforms. You shouldn't use IPv6 in
 # your configuration files unless you have IPv6 installed.
 #
+# As a special case, `LISTEN * <port>` (with an asterisk) would try
+# to listen on "ANY" IP address for both IPv4 (0.0.0.0) and IPv6 (::0),
+# subject to `upsd` command-line arguments or OS/kernel configuration.
+# If the system supports "dual-stack" mode (IPv4-mapped IPv6) per RFC-3493
+# then there would be one listening socket for both address types.
+#
```

Comment:
I guess it's special, but * for INADDR_ANY is normal.

This says "would", but samples should describe the present.    Here, I would just say 

```
# `LISTEN * <port>` will cause `upsd` to listen on both IPv6 (::0) and IPv4 (0.0.0.0), subject to
# system configuration.
```

I don't think this needs to explain that if your system has a mechanism to do this with one socket that nut will use that mechanism.

Also, I changed OS/kernel to system, because system is less abbreviatish, and because kernel is a component of the operating system so OS/kernel is odd to me.

Reply:
Thanks for the suggestions :)

As for "normal", I stumbled upon the problem because apparently my `LISTEN *` only bound to IPv4, and the only input for that was the system's `getaddrinfo()` to resolve the asterisk.

As for OS and kernel as its part... nowadays there are quite a few projects with essentially same userland (and its config toggles and defaults) vs. the actual kernel underneath - see containers, abstractions line cygwin/msys2/mingw, etc. In some cases the kernel would not implement the mapped addresses (like OpenBSD apparently, due to security-first) and so no system or setsockopt toggles would help. In other cases there is a useful system-level toggle which defaults to one of the choices, and NUT can meaningfully ask to configure a particular socket one way or another.

Reply:
Yes, using getaddrinfo for * was a bug, and it's good we are fixing this.

My point is that "operating system" is an object constructed of a userland and a kernel, and that overall object has a behavior.   A kernel can have defaults, and the userland at boot can write sysctls to adjust it.  Or all sorts of other things.   Nut need not concern itself with how operating systems are assembled.   So "operating system" is the term that refers to the union of the system components and their configuration.    The use of kernel in contexts like this is just an artifact of the historically strange situation where Linux is a kernel only and there are "GNU/Linux distributions".

As for OpenBSD, I can completely see them removing support for mapped addresses.   But it isn't that the OpenBSD kernel removed them and the OpenBSD userland did not; it's that the OpenBSD operating system did.    I would expect the defines to be gone from /usr/include, and the man pages to be updated.    Or maybe they left the defines and have the ioctl still be it always returns an error, and the man pages say this.   But  it's the operating system we are dealing with.